### DISCUSSION TITLE
`hasThreadgate` viewer state on posts

### DISCUSSION THREAD

**Discussion by mary-ext** [bluesky-social/atproto#1899] [Bluesky Lexicons]
**Is your feature request related to a problem? Please describe.**

Right now, apps can't show any indication of there actually being a threadgate in place until it finishes retrieving the threadgate record (if it even exists)

The `canReply` bool isn't useful for this as ideally apps should also show the indication on users who are able to reply.

**Describe the solution you'd like**

A `hasThreadgate` viewer state, ideally linking to the record URI of the threadgate

**Describe alternatives you've considered**

Always retrieve the threadgate after retrieving the post thread, the problem is that this would push existing content around (UX issue), one way to fix this would be to block page rendering entirely, but this can be problematic with unstable networks

**Additional context**

None.

**Comment by bnewbold** [comment]
(generally try to move this type of design question from "Issues" to "Discussions")

@dholms any thoughts on this one?

**Reply by mary-ext** [reply]
sorry ^^ wasn't aware of this, it's a bit tempting to just create an issue especially since the feature request template is right there haha.

after thinking on it some more, it might honestly be worth to reintroduce the `threadViewerState` field but this time containing the actual threadgate record, since this is something that has to be fetched anyway by any clients that intends to show these threads properly.

but we'll definitely make do with just having some indication that there's a threadgate being applied at all, not sure how the official app intends to go around doing this.

**Reply by dholms** [reply]
Yup as Devin pointed out, the threadgate view exists on the root post, which is hydrated into both thread views & feed views

**Comment by devinivy** [comment]
You should be able to find the threadgate record on the root post of the thread.  It also has hydrated information about any lists referenced by the threadgate.  The lexicon for this lives here: https://github.com/bluesky-social/atproto/blob/7edad62c12038de09452f8e005be5a1a75e18873/lexicons/app/bsky/feed/defs.json#L34

**Reply by mary-ext** [reply]
ahh I actually hadn't noticed this at all, thank you!
